### PR TITLE
Hooks logic update and more

### PR DESCRIPTION
**Pull request by armanist**

### CODE REVIEW

**Review by andrey-smaelov — APPROVED** [review]

File: src/Helpers/url.php [hunk 1]
```diff
@@ -36,9 +36,9 @@ function current_url(): string
 /**
  * Redirect
  * @param string $url
- * @param integer|null $code
+ * @param integer $code
```
Comment (Vahram1995) [inline src/Helpers/url.php:39]: integer should be int

File: src/Http/Response/HttpResponse.php [hunk 2]
```diff
@@ -176,14 +177,10 @@ public static function getStatusText(): string
      * @param int|null $code
```
Comment (Vahram1995) [inline src/Http/Response/HttpResponse.php:177]: remove null

File: src/Http/Response/HttpResponse.php [hunk 3]
```diff
@@ -209,12 +206,12 @@ public static function json(array $data = null, int $code = null)
 
     /**
      * Prepares the JSONP response
-     * @param array|null $data
      * @param string|null $callback
```
Comment (Vahram1995) [inline src/Http/Response/HttpResponse.php:209]: remove null

File: tests/unit/Helpers/HelperTest.php [hunk 4]
```diff
@@ -218,7 +220,8 @@ public function testMvcHelpers()
 
         $this->assertEquals('GET', route_method());
 
-        $this->assertEquals('api-user/12', route_uri());;
+        $this->assertEquals('api-user/12', route_uri());
+        ;
```
Comment (Vahram1995) [inline tests/unit/Helpers/HelperTest.php:224]: remove semicolon

File: tests/unit/Http/ResponseTest.php [hunk 5]
```diff
@@ -154,16 +155,30 @@ public function testResponseJsonContent()
...
 
-            $response->json([
-                'gender' => 'male',
-                'role' => 'user'
-            ], 200);
+            $response->json(
+                    [
+                        'gender' => 'male',
+                        'role' => 'user'
+                    ],);
```
Comment (Vahram1995) [inline tests/unit/Http/ResponseTest.php:162]: remove comma

**Review by Vahram1995 — CHANGES_REQUESTED** [review]
please check comments

**Review by Vahram1995 — APPROVED** [review]

**Pull request merged**